cmake/FindPTLS.cmake

```diff
@@ -0,0 +1,29 @@
+# - Try to find Picotls
+
+find_path(PICOTLS_INCLUDE_DIR
+    NAMES picotls.h picotls/openssl.h picotls/minicrypto.h
+    HINTS ${CMAKE_SOURCE_DIR}/../picotls/include )
+
+MESSAGE (STATUS "found picotls.h at ${PICOTLS_INCLUDE_DIR}" )
+
+FIND_LIBRARY(PTLS_CORE picotls-core
+    HINTS $(CMAKE_BINARY_DIR)/../picotls
+           ../picotls
+)
+
+MESSAGE(STATUS "Found picotls-core at : ${PTLS_CORE}" )
+
+FIND_LIBRARY(PTLS_MINICRYPTO picotls-minicrypto
+    HINTS $(CMAKE_BINARY_DIR)/../picotls
+          ../picotls
+)
+
+MESSAGE(STATUS "Found picotls-crypto at : ${PTLS_MINICRYPTO}" )
+
+FIND_LIBRARY(PTLS_OPENSSL picotls-openssl
+    HINTS $(CMAKE_BINARY_DIR)/../picotls
+          ../picotls
+)
+
+MESSAGE(STATUS "Found picotls-openssl at : ${PTLS_OPENSSL}" )
+
```

Comment:
Consider calling FPHSA which takes care of setting the required variables (such as `PTLS_FOUND`) which is used by `find_package`. Conventionally, use `PTLS_INCLUDE_DIRS` and `PTLS_LIBRARIES` to export a list when both the include dir and libraries are found. The whole `FindPTLS.cmake` file will roughly look like this (untested):

```cmake
find_path(PICOTLS_INCLUDE_DIR picotls.h HINTS ${CMAKE_SOURCE_DIR}/../include)

set(PTLS_HINTS ${CMAKE_BINARY_DIR}/../picotls ../picotls)

find_package(PTLS_CORE_LIBRARY picotls-core HINTS ${PTLS_HINTS})
find_package(PTLS_MINICRYPTO_LIBRARY picotls-minicrypto HINTS ${PTLS_HINTS})
find_package(PTLS_OPENSSL_LIBRARY picotls-openssl HINTS ${PTLS_HINTS})

find_package_handle_standard_args(PTLS REQUIRED_VARS
                                  PTLS_CORE_LIBRARY PTLS_MINICRYPTO_LIBRARY PTLS_OPENSSL_LIBRARY PICOTLS_INCLUDE_DIR)
if(PTLS_FOUND)
    set(PTLS_LIBRARIES ${PTLS_CORE_LIBRARY} ${PTLS_MINICRYPTO_LIBRARY} ${PTLS_OPENSSL_LIBRARY})
    set(PTLS_INCLUDE_DIRS ${PTLS_INCLUDE_DIR})
endif()

mark_as_advanced(PTLS_LIBRARIES PTLS_INCLUDE_DIRS)
```

Reply:
Do I need to somehow load a package for FPHSA? I am getting an error:
```
CMake Error at cmake/FindPTLS.cmake:13 (find_package_handle_standard_args):
  Unknown CMake command "find_package_handle_standard_args".
Call Stack (most recent call first):
  CMakeLists.txt:11 (find_package)
```

Reply:
OK, I need to add:
```
include(FindPackageHandleStandardArgs)
```
Also, the command to find the library has to be `find_library` instead of `find_package`. But with that, it works!